app/services/moves/updater.rb

```diff
@@ -15,6 +15,7 @@ def call
       self.status_changed = move.status_changed?
 
       move.save!
+      move.allocation&.refresh_moves_count! if status_changed
```

Comment:
instead of attempting to save changes twice, is there a nice way we can include the `moves_count` update, maybe in a `after_validation` hook?

Reply:
This saves the move once and the allocation once - I also wanted to avoid needlessly updating the allocation if the status hasn't changed. I prefer this approach as it's explicit and simple, plus we can see all of the steps involved in updating a move in one place (here in the service).

Reply:
My main concern lies in ensuring data consistency. When we perform two saves and one of them fails, we might get into an inconsistent state. One way around this would be to do something similar to https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/487/files#diff-ef7828cdbd6b61227ed9a64910384e69R65 where we still keep the `refresh_moves_count` in the allocation, but it only sets the attribute and does not save the record:
```
move.assign_attributes(attributes)
self.status_changed = move.status_changed?
move.allocation&.refresh_moves_count if status_changed
move.save!
```
This way we perform all our updates in one transaction.
Or we could wrap this whole thing in a transaction, I prefer the former though.

Reply:
Added a transaction wrapper so that the state is consistent even if saving the allocation somehow fails after saving the move.